ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -472,11 +472,14 @@ func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interfac
 		return dataMap, err
 	}
 
-	// merge the pre-defined manifests
-	dataMap, err = MergeManifests(dataMap, doNotMerge)
-	if err != nil {
-		log.Printf("Error could not merge extra-manifest %s.%s %s\n", clusterSpec.ClusterName, clusterSpec.ExtraManifestPath, err)
-		return dataMap, err
+	// check if legacy config is enabled to stop splitting
+	if clusterSpec.LegacyConfig == nil || !clusterSpec.LegacyConfig.SplitMachineConfigCRs {
```

Comment:
In this case we want the default behavior (if the user does not give this option) to be _not_ merging the CRs. The merging feature should be an opt-in where the user has to explicitly enable it.

Reply:
Flipped the logic